File: tests/fixtures/codegen/expected/typescript/runtime/queries/admin/events/GetAliases.ts
import type { GetAliasesParams, GetAliasesRow } from "../../../../types/queries/admin/events/GetAliases.js";
export type { GetAliasesParams, GetAliasesRow } from "../../../../types/queries/admin/events/GetAliases.js";
import type * as pgsid from "./pgsid/queryable.js";
export const getAliasesSql = "SELECT p.id AS public_id, q.id AS dotted_id, r.id AS underscored_id\nFROM public.events p, \"billing.extra\".events q, billing_extra.events r;";
export async function getAliases(db: pgsid.Queryable): Promise<GetAliasesRow | undefined> {
    const result = await db.query(getAliasesSql, []);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as GetAliasesRow;
}

File: tests/fixtures/codegen/expected/go/queries/admin/events/getaliases.go
package events

import (
	"context"
	billing_extra2 "example.com/pgsid-fixture/generated/go/schema/billing.extra"
	billing_extra "example.com/pgsid-fixture/generated/go/schema/billing_extra"
	public "example.com/pgsid-fixture/generated/go/schema/public"
)

const GetAliasesSQL = "SELECT p.id AS public_id, q.id AS dotted_id, r.id AS underscored_id\nFROM public.events p, \"billing.extra\".events q, billing_extra.events r;"

type GetAliasesParams struct {
}
type GetAliasesRow struct {
	PublicId      public.EventId         `db:"public_id"`
	DottedId      billing_extra2.EventId `db:"dotted_id"`
	UnderscoredId billing_extra.EventId  `db:"underscored_id"`
}

func (q *Queries) GetAliases(ctx context.Context) (GetAliasesRow, error) {
	var row GetAliasesRow
	err := q.db.QueryRow(ctx, GetAliasesSQL).Scan(&row.PublicId, &row.DottedId, &row.UnderscoredId)
	return row, err
}
